src/Pagination.js

```diff
@@ -0,0 +1,313 @@
+import React from 'react';
+
+import PropTypes from 'prop-types';
+
+let _ = require('lodash');
+const classNames = require('classnames');
+
+class Pagination extends React.Component {
+  state = {
+    arrPages: [],
+    nextDisabled: false,
+    prevDisable: true,
+    withInfo: false,
+    changedView: false,
+  }
+
+  componentDidMount() {
+    const { page, perPage, total } = this.props;
+
+    this.setState({
+      arrPages: Array(Math.ceil(total / perPage)).fill(1).map((p, index) => {
+        if (index === page) {
+          return {
+            additionalInfo: `${index}-${perPage}`,
+            active: true,
+          };
+        }
+
+        return {
+          additionalInfo: `${(index * perPage)}-${((index + 1) * perPage) > total
+            ? total
+            : ((index + 1) * perPage)}`,
+          active: false,
+        };
+      }),
+    });
+  }
+
+  isUnDisable = (page) => {
+    const { onPageChange, preparedArrPages } = this.props;
+
+    onPageChange(page);
```

Comment:
you can't have two setstate inside one method, no need to create this func

Reply:
I need to follow on change page